lib/linguist/heuristics.rb

```diff
@@ -17,9 +17,8 @@ def self.call(blob, candidates)
       data = blob.data
 
       @heuristics.each do |heuristic|
-        if heuristic.matches?(blob.name)
-          languages = Array(heuristic.call(data))
-          return languages if languages.any? || languages.all? { |l| candidates.include?(l) }
```

Comment:
Any reason why you removed the second check (`languages.all? { |l| candidates.include?(l) }`)? This is in place to make sure heuristic rule don't return new languages we've already cross off the list.